components/dashboard/src/components/PrebuildLogs.tsx

```diff
@@ -0,0 +1,205 @@
+        // are still preparing for launch. This means we're building the Docker image for the workspace.
+        case "preparing":
+          service.server.watchWorkspaceImageBuildLogs(workspace!.id);
+          break;
+
+        // Pending means the workspace does not yet consume resources in the cluster, but rather is looking for
+        // some space within the cluster. If for example the cluster needs to scale up to accomodate the
+        // workspace, the workspace will be in Pending state until that happened.
+        case "pending":
+          break;
+
+        // Creating means the workspace is currently being created. That includes downloading the images required
+        // to run the workspace over the network. The time spent in this phase varies widely and depends on the current
+        // network speed, image size and cache states.
+        case "creating":
+          break;
+
+        // Initializing is the phase in which the workspace is executing the appropriate workspace initializer (e.g. Git
+        // clone or backup download). After this phase one can expect the workspace to either be Running or Failed.
+        case "initializing":
+          break;
+
+        // Running means the workspace is able to actively perform work, either by serving a user through Theia,
+        // or as a headless workspace.
+        case "running":
+          break;
+
+        // Interrupted is an exceptional state where the container should be running but is temporarily unavailable.
+        // When in this state, we expect it to become running or stopping anytime soon.
+        case "interrupted":
+          break;
+
+        // Stopping means that the workspace is currently shutting down. It could go to stopped every moment.
+        case "stopping":
+          break;
+
+        // Stopped means the workspace ended regularly because it was shut down.
+        case "stopped":
+          service.server.watchWorkspaceImageBuildLogs(workspace!.id);
+          break;
+    }
+    if (workspaceInstance?.status.conditions.failed) {
+      setError(new Error(workspaceInstance.status.conditions.failed));
+    }
+  }, [ workspaceInstance?.status.phase ]);
+
+  return <>
+    <div className="capitalize">{workspaceInstance?.status.phase}</div>
+    <Suspense fallback={<div />}>
+      <WorkspaceLogs classes="h-64 w-full" logsEmitter={logsEmitter} errorMessage={error?.message} />
+    </Suspense>
+    <div className="mt-2 flex justify-center space-x-2">
+      {workspaceInstance?.status.phase === 'stopped'
+        ? <a href={workspace?.contextURL ? '/#' + workspace.contextURL.replace(/^prebuild/, '') : undefined}><button>Open Workspace</button></a>
+        : <button className="secondary disabled" disabled={true}>Open Workspace</button> }
+    </div>
+  </>;
+}
+
+export function watchHeadlessLogs(server: GitpodServer, instanceId: string, onLog: (chunk: string) => void, checkIsDone: () => Promise<boolean>): DisposableCollection {
```

Comment:
Note: I'm moving this function out of `WorkspaceLogs.tsx` and into the new component `PrebuildLogs.tsx`, because it is imported eagerly in various places.

However, we don't want to import `WorkspaceLogs.tsx` eagerly, because that file imports all of Xterm.js (you'll notice that `WorkspaceLogs` itself is always imported lazily -- same for `MonacoEditor.tsx` now).